### PR TITLE
BUG: sort_index throws IndexError for some permutations (#26053)

### LINKED ISSUE #26053: sort_index throws IndexError for some permutations
#### Code Sample

```python
df = pd.DataFrame(list(range(8)))
a.index = pd.MultiIndex.from_product([[0,1]]*3)
a.sort_index(level=[1,2,0])
```
#### Problem description

This throws an `IndexEror: pop index out of range`. 
If I change the order to `[2,0,1]`, it doesn't.

python: 3.6.6.final.0
pandas: 0.24.2
numpy: 1.15.4

### PR DESCRIPTION
**Pull request by jayanthkaturi**
- [x] closes #26053 
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] tests added / passed

### CODE REVIEW

File: pandas/tests/frame/test_sorting.py [hunk 1]
```diff
@@ -496,12 +496,22 @@ def test_sort_index_duplicates(self):
...
         mi = MultiIndex.from_tuples([[1, 1, 3], [1, 1, 1]], names=list('ABC'))
         df = DataFrame([[1, 2], [3, 4]], mi)
+
         res = df.sort_index(level='A', sort_remaining=False)
         assert_frame_equal(df, res)
 
         res = df.sort_index(level=['A', 'B'], sort_remaining=False)
         assert_frame_equal(df, res)
 
+        res = df.sort_index(level=['C', 'B', 'A'])
```
Comment (jreback) [inline pandas/tests/frame/test_sorting.py]: can you put a comment here with the issue number

File: pandas/tests/frame/test_sorting.py [hunk 2]
```diff
@@ -496,12 +496,22 @@ def test_sort_index_duplicates(self):
...
+        assert_frame_equal(df.iloc[[1, 0]], res)
```
Comment (jreback) [inline pandas/tests/frame/test_sorting.py]: write these as

expected = ....
assert_frame_equal(result, expected)

need also to fix the others in this test

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/multi.py [hunk 3]
```diff
@@ -2085,8 +2085,22 @@ def sortlevel(self, level=0, ascending=True, sort_remaining=True):
             shape = list(self.levshape)
 
             # partition codes and shape
-            primary = tuple(codes.pop(lev - i) for i, lev in enumerate(level))
-            primshp = tuple(shape.pop(lev - i) for i, lev in enumerate(level))
+            primary = []
+            primshp = []
+
+            # Add the required codes and shape
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: write both of these as a list comprehensiion (as was original, just w/o the pop)

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]
is the original issue a duplicate of what you have here (if it is, that's ok), otherwise pls add it.

pls add a what's new note in Bug fixes, reshaping section

**Comment by jayanthkaturi** [comment]
this is same/duplicate as the original issue. I updated the PR, pls let me know if anything else is required. Thanks!

File: doc/source/whatsnew/v0.25.0.rst [hunk 4]
```diff
@@ -405,6 +405,7 @@ Reshaping
 - Bug in :func:`merge` where merging with equivalent Categorical dtypes was raising an error (:issue:`22501`)
 - Bug in :class:`DataFrame` constructor when passing non-empty tuples would cause a segmentation fault (:issue:`25691`)
 - Bug in :func:`pandas.cut` where large bins could incorrectly raise an error due to an integer overflow (:issue:`26045`)
+- Bug in :func:`DataFrame.sort_index` where sorting with the first index level last was throwing an error (:issue:`26053`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: this is not giving me any idea what you are changing, can you reword a bit

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.25.0.rst (hunk 4 above)
Comment (jayanthkaturi) [reply doc/source/whatsnew/v0.25.0.rst]: > Bug in :func:`DataFrame.sort_index` where an error is thrown when a multi-indexed 
> DataFrame is sorted on all levels with the initial level sorted last (:issue:`26053`)
is this one better?

Comment (jreback) [reply doc/source/whatsnew/v0.25.0.rst]: sure

**Comment by jreback** [comment]
also pls merge master

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jayanthkaturi

**Pull request merged**